Maskbook/Wallet/WalletService/PendTransacation/PendTransactionManager.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  PendTransactionManager.swift
+//  Maskbook
+//
+//  Created by caiyu on 2022/4/7.
+//  Copyright © 2022 dimension. All rights reserved.
+//
+
+import Foundation
+import web3swift
+import Combine
+import BigInt
+
+class PendTransactionManager {
+
+    private var pendList:[PendTransactionModel] = []
+    private var timer: Timer?
+    var subscriptions: Set<AnyCancellable> = []
+    public static let shared = PendTransactionManager()
+    
+    public var pendTransactions: CurrentValueSubject<[PendTransactionModel], Never> = CurrentValueSubject([])
+
+    private func start() {
+            timer = Timer.scheduledTimer(withTimeInterval: 5, repeats: true) { _ in
+                DispatchQueue.global().async {
+                    guard let web3Provier = Web3ProviderFactory.provider?.eth else {
+                        self.timer?.invalidate()
+                        return
+                    }
+                    
+                    if self.pendList.isEmpty {
+                        self.timer?.invalidate()
+                    }
+                    
+                    for (index, pendingTranscation) in self.pendList.enumerated() {
+                        web3Provier.getTransactionReceiptPromise(pendingTranscation.txHash).done { transactionReceipt in
+                            if transactionReceipt.status != .notYetProcessed {
+                                self.pendList.remove(at: index)
+                            }
+                            self.pendTransactions.send(self.pendList)
```

Comment:
It's better if we could have a `PassThroughSubject` that sends the status change events of pending TXs.
Like:
```
var pendingTxFinishEvents = PassthroughSubject<PendTransactionModel, ConfimState>()
```
Once we found a pending tx finishes (either confirmed or failed), we:
1. modify the pendingTxList.
2. We emit the event with the tx and the latest status.